src/history/HistoryManagerImpl.cpp

```diff
@@ -147,6 +148,12 @@ HistoryManagerImpl::logAndUpdatePublishStatus()
 size_t
 HistoryManagerImpl::publishQueueLength() const
```

Comment:
General question on this class: it looks like right now the choice between throwing and simulating function behavior is sort of arbitrary. For example, we allow to "publish" queued history, or to check publish queue length, but throw if we're trying to queue something up. It'd be nice to have consistent behavior across functions (note for the future: maybe at some point we can even not spin up HistoryManager at all when in "publishing disabled" mode, i think it's centrally controlled in LedgerManager anyway)

Reply:
Unfortunately HistoryManager provides also the interface we use to do ledger-range calculations (based on whether or not we're in accelerated-time mode) and a few other query-history helpers, so there are several dozen call sites even in not-publishing mode. Again, I'm not strictly opposed to eliminating these / trying to move to a no-history-manager mode, but it involves yet more code motion and the PR is already pretty big.